### PR TITLE
Toni Baker-View User Profiles if Admin 

### PR DESCRIPTION
**Pull request by ToniJBaker**
Trello Ticket # 23 View All UserProfiles
https://trello.com/c/joWtLylC/23-23-view-all-user-profiles

Seed Data: 
https://github.com/NewForce-Cohort-6/tabloidfullstack-stuffing/blob/main/SQL/01_Tabloid_Create_DB.sql

https://github.com/NewForce-Cohort-6/tabloidfullstack-stuffing/blob/main/SQL/02_Tabloid_Seed_Data.sql

Modifications:
VS:
UserRepository Line 13-49 GellAll();
IUserRepository Line 10 List<UserProfile> GetAll();
UserProfileController Line 20-25 [HTTP GET]

VS Code:
UserProfileManager Line 26-30 getCurrentUser();  and Line 47-50 getAllUsers();
Header.js lines 39-48 Nav Element for User Profiles, conditional for admin or not line 19-22 useEffect and line 17 useState
new folder-users
new file UserProfileItem.js
new file UserProfiles.js

1. Instructions: Log in as foo@bar.com (admin profile) 
2. You should see the User Profiles nav element in the header to select
3. Click on User Profiles
4. Table should appear listing users by Name, Display Name and User Type
5. Logout
6. Login as rdo1@timesonline.co.ukx you should not see the user profile link in the header.
7. logout

### CODE REVIEW

**Comment by shanedbutler** [comment]
I've tested if a non-admin (author) user can access the post list. A few things:

- I've confirmed that a non-admin cannot see the header navbar item that leads to `/users`
- However, a non-admin can access the `/users` page by directly typing in that path into the URL bar

This may not be something that needs to be taken care of now, but something to be aware of.

File: Tabloid/client/src/components/users/UserProfiles.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  const getUserProfiles = () =>{
+    getAllUsers().then(allUsers => setUserProfiles(allUsers))
+  }
+  
+  useEffect(() => {
+        getUserProfiles();
+    }, []); 
+  
+  return (
+    <Table>
```
Comment (shanedbutler) [inline Tabloid/client/src/components/users/UserProfiles.js:20]: `<Table></Table>` can be wrapped with `<div className="m-5"></div>`

This is the bootstrap class property outlined in the Google Doc and provides the margin of 5 all around.

See the code in `PostList.js` for an example.

**Review by shanedbutler — CHANGES_REQUESTED** [review]
Please add margins as in above comment to your `<Table>`

Otherwise looks good and is tested working!

**Review by shanedbutler — APPROVED** [review]
Looks good!

**Review by robert-brunner — APPROVED** [review]
It works.

**Pull request merged**